### PR TITLE
More effective and precise snapper guide lines (crosshairs)

### PR DESCRIPTION
**Pull request by Osndok**
* Crosshairs vanish around snapPoint
* Cursor vanishers around snapPoint

### CODE REVIEW

**Comment by Osndok** [comment]
librecad-interface-improvements-screenshot-n1

File: librecad/src/lib/actions/rs_snapper.cpp [hunk 1]
```diff
@@ -758,22 +771,56 @@ void RS_Snapper::drawSnapper()
...
-						{{0., graphicView->toGuiY(pImpData->snapCoord.y)},
-						 {double(graphicView->getWidth()),
-						  graphicView->toGuiY(pImpData->snapCoord.y)}
-                                                                });
-						line->setPen(pImpData->line_pen);
-                        container->addEntity(line);
+                        RS_OverlayLine *line;
+
+						// ----O     (Left)
+						line=new RS_OverlayLine(nullptr, {
```
Comment (dxli) [inline librecad/src/lib/actions/rs_snapper.cpp]: is there a memory leak? for the pointer line?

Comment (Osndok) [inline librecad/src/lib/actions/rs_snapper.cpp]: Of that, I am not sure. However, my patch uses the same convention that the former code used... every line is added to the container. So... I would think that if my patch leaks 4 lines, then they original code leaked 2.

Comment (r-a-v-a-s) [inline librecad/src/lib/actions/rs_snapper.cpp]: I just found:
http://wiki.qt.io/Profiling_and_Memory_Checking_Tools
http://doc.qt.io/qtcreator/creator-analyzer.html

Although I'm busy with other things.

Comment (r-a-v-a-s) [inline librecad/src/lib/actions/rs_snapper.cpp]: As long as the entity container is deleted (and it wasn't https://github.com/LibreCAD/LibreCAD/commit/bd939eaeb0f17802f67bda330311229bded94475)
`RS_EntityContainer::clear` deletes the lines.

Comment (dxli) [inline librecad/src/lib/actions/rs_snapper.cpp]: @Osndok @r-a-v-a-s 

sorry, my bad.

at line 710 of RS_Snapper, the "Snapper" container is cleared

```C++
    graphicView->getOverlayContainer(RS2::Snapper)->clear();
```
and this container is created by default autoDelete=true. Therefore, there is no memory leak.

File: librecad/src/lib/actions/rs_snapper.cpp [hunk 2]
```diff
@@ -758,22 +771,56 @@ void RS_Snapper::drawSnapper()
                 }
                 else //orthogonal crosshair
                 {
+					double snapX=graphicView->toGuiX(pImpData->snapCoord.x);
+					double snapY=graphicView->toGuiY(pImpData->snapCoord.y);
+
+					double viewWidth=double(graphicView->getWidth());
```
Comment (dxli) [inline librecad/src/lib/actions/rs_snapper.cpp]: I prefer default implicit type conversion:

```C++
double viewWidth = graphicView->getWidth();
double viewWidth{graphicView->getWidth()};
```

Comment (Osndok) [inline librecad/src/lib/actions/rs_snapper.cpp]: Technically, this does not actually originate from me, but from the original line 765, as you can see below when I was factoring out the expression. Do you want me to push another commit to change this?

**Comment by dxli** [comment]
@Osndok 

looks like more work is needed for isometric grids (Settings -> Current Drawing Preferences -> Grid -> isometric)

**Comment by Osndok** [comment]
Ahh, yes... my observation is that it is not broken, but that the cross hairs do not stand off from the snap point, do you concur?

**Comment by Osndok** [comment]
I tried to dive in and fix this, but the master branch does not compile for me at the moment:

<pre>
ui/forms/qg_dlgoptionsgeneral.cpp: In member function ‘void QG_DlgOptionsGeneral::init()’:
ui/forms/qg_dlgoptionsgeneral.cpp:115:31: error: ‘class QComboBox’ has no member named ‘setCurrentText’
     indicator_lines_combobox->setCurrentText(indicator_lines_type);
                               ^
ui/forms/qg_dlgoptionsgeneral.cpp:121:31: error: ‘class QComboBox’ has no member named ‘setCurrentText’
     indicator_shape_combobox->setCurrentText(indicator_shape_type);
                               ^
make[2]: *** [../../generated/librecad/obj/qg_dlgoptionsgeneral.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/mnt/projects/LibreCAD/librecad/src'
make[1]: *** [sub-src-make_default] Error 2
make[1]: Leaving directory `/mnt/projects/LibreCAD/librecad'
make: *** [sub-librecad-make_default-ordered] Error 2
</pre>

I'm not sure what's going on, but found others have experienced it too:

https://forum.qt.io/topic/5120/qcombobox-setcurrenttext
https://forum.qt.io/topic/45373/a-question-about-qcombobox-setcurrentindex

**Comment by r-a-v-a-s** [comment]
Did you run Qmake?

**Comment by r-a-v-a-s** [comment]
Looks like `setCurrentText` is not in Qt 4.

**Comment by r-a-v-a-s** [comment]
Try it with https://github.com/LibreCAD/LibreCAD/commit/6f6ab1c21f771b286a8650c4bafab1823d267699

**Comment by Osndok** [comment]
Thanks, ravas... it looks like that fixes the build, but I'm out of time for tonight, so I'll have to come back to this later.
